administrator/components/com_associations/views/association/view.html.php

```diff
@@ -93,32 +93,36 @@ public function display($tpl = null)
 		}
 
 		$this->app = JFactory::getApplication();
-		$input = $this->app->input;
-		$assoc = JLanguageAssociations::isEnabled();
 
-		$referenceId = $input->get('id', '0');
-		$associatedComponent = $input->get('acomponent', '');
-		$associatedView = $input->get('aview', '');
-
-		$this->link = "";
-
-		if ($associatedComponent == 'com_categories')
-		{
-			// If it's categories
-			$this->link = 'index.php?option=' . $associatedComponent . '&task=category.edit&layout=modal&tmpl=component&id='
-				. $referenceId . '&extension=' . $associatedView;
-		}
-		elseif ($associatedComponent == 'com_menus')
+		$this->form  = $this->get('Form');
+		$input     = $this->app->input;
+		$assoc     = JLanguageAssociations::isEnabled();
+		
+		$formData = new JInput($input->get('jform', '', 'array'));
+
+		$associatedComponent = $input->get('acomponent', '', 'string');
+		$associatedView      = $input->get('aview', '', 'string');
+
+		$options = array(
+			'option'    => $associatedComponent,
+			'view'      => $associatedView,
+			'task'      => $associatedView . '.edit',
+			'layout'    => 'edit',
+			'tmpl'      => 'component',
```

Comment:
@andrepereiradasilva it already is tmpl = component, what I meant was, how do I save the item without loading the whole thing? i.e., save + load with tmpl=component. Should I save and reload the iframe? is this a good approach?